### ISSUE TITLE
Use across assemblies that use `[assembly: InternalsVisibleTo(...)]` produces warnings

### ISSUE THREAD

**Issue by rickbrew** [k94ll13nn3/AutoConstructor#93]
Let's say you have assembly A and assembly B, with B referencing A.

Assembly A declares `[assembly: InternalsVisibleTo("B")]`. Both assemblies make use of `[AutoConstructor]`.

You'll get this warning when compiling assembly B:
```
warning CS0436: The type 'AutoConstructorAttribute' in 'AutoConstructor.Generator\AutoConstructor.Generator.AutoConstructorGenerator\AutoConstructorAttribute.cs' conflicts with the imported type 'AutoConstructorAttribute' in 'A, Version=1.2.3.4, Culture=neutral, PublicKeyToken=null'. Using the type defined in 'AutoConstructor.Generator\AutoConstructor.Generator.AutoConstructorGenerator\AutoConstructorAttribute.cs'.
```

I compile with warnings-as-errors so this actually prevents me from compiling. Thankfully you can suppress this warning, which also prevents it from turning into an error, via `.editorconfig` (or elsewhere) and then the compiler is happy:
```
# CS0436: Type conflicts with imported type
# Ensures that per-assembly source-generated attributes, e.g. [AutoConstructor], will not raise errors
dotnet_diagnostic.CS0436.severity = none
```

@Sergio0694 just fixed this exact issue in a new generator by emitting the suppression automatically and only for the type in question.

**Comment by Sergio0694** [comment]
@k94ll13nn3 if it helps, you should be able to pretty much just copy [this](https://github.com/Sergio0694/privatelike/blob/9410c0179e880cb5a679196890df5625e95fad06/src/PrivateLike/PrivateLikeAttributeDiagnosticSuppressor.cs#L11), minus some tweaks 🙂

**Comment by k94ll13nn3** [comment]
Thanks for the report and for the sample code.
Fixed in 5.2.0, should be out on NuGet shortly.
